lca-cli/ibi-preparation/ibipreparation.go

```diff
@@ -211,3 +235,45 @@ func (i *IBIPrepare) cleanupRhcosSysroot() error {
 	}
 	return nil
 }
+
+func (i *IBIPrepare) shouldOverrideSeedRegistry(seedMirrorRegistryConfigured bool, seedReleaseRegistry string) (bool, error) {
```

Comment:
Its already implemented at https://github.com/openshift-kni/lifecycle-agent/blob/e8a8907961f834444fc215a9ea19e5015179f0c5/utils/client_helper.go#L414

Reply:
Same as https://github.com/openshift-kni/lifecycle-agent/pull/727/files#r1935308620.

Reply:
Hmm, still will be nice to try to share it, seems like it can get targetMirroredRegistries from outside and in this case logic is pretty the same or i missed something?

Reply:
I think both your suggestions look a lot better, it's now less confusing :+1:  

@tsorya I don't think you're missing anything, the difference is where we get the target's mirror registry sources from, either via the cluster's kube API or the `registries.conf` file.